File: LongChallenge_2018_10_BITOBYT.py
# code_report Solution
# https://youtu.be/EbgOk_IDl1I

t = int(input())
for _ in range(t):
    l = [1, 0, 0]
    n = int(input())
    for i in range(3, n+1):
        d = i % 26
        if d == 3:  l[0], l[1] = l[1], l[0]
        if d == 11: l[1], l[2] = l[2], l[1]
        if d == 1:  l[0], l[2] = l[2] * 2, l[0]
    print(' '.join(map(str, l)) )

File: LongChallenge_2018_10_BITOBYT.cpp
// code_report Solution
// https://youtu.be/EbgOk_IDl1I

#include <iostream>
#include <vector>
#include <algorithm> 

using namespace std;

int main() {
   int t; cin >> t;
   while (t--) {
      vector<long long> v = { 1, 0, 0 };
      int n; cin >> n;
      for (int i = 3; i <= n; ++i) {
         int d = i % 26;
         if (d == 3)  swap(v[1], v[0]);
         if (d == 11) swap(v[2], v[1]);
         if (d == 1)  swap(v[0], v[2]), v[0] *= 2;
      }
      cout << v[0] << ' ' << v[1] << ' ' << v[2] << endl;
   }
   return 0;
}
